File: app/src/main/java/com/inelasticcollision/recipelink/data/local/converter/DateTypeConverter.java
package com.inelasticcollision.recipelink.data.local.converter;

import androidx.annotation.NonNull;
import androidx.room.TypeConverter;

import java.util.Date;

public class DateTypeConverter {

    @NonNull
    @TypeConverter
    public static Date fromTimestamp(@NonNull Long value) {
        return new Date(value);
    }

    @NonNull
    @TypeConverter
    public static Long dateToTimestamp(@NonNull Date date) {
        return date.getTime();
    }

}

File: app/src/main/java/com/inelasticcollision/recipelink/data/cache/converter/DateTypeConverter.kt
package com.inelasticcollision.recipelink.data.cache.converter

import androidx.room.TypeConverter
import java.util.*

object DateTypeConverter {

    @JvmStatic
    @TypeConverter
    fun fromDate(date: Date): Long {
        return date.time
    }

    @JvmStatic
    @TypeConverter
    fun toDate(dateLong: Long): Date {
        return Date(dateLong)
    }
}